### DISCUSSION TITLE
What does hyper use to parse HTTP2 headers?

### DISCUSSION THREAD

**Discussion by thesw4rm** [hyperium/hyper#2825] [Q&A] [answered]
I'm aware that httparse is used to parse for HTTP 1.0 and HTTP 1.1. Is there a custom parser implemented for HTTP2, or is it a separate project? httparse does not seem to support HTTP2 for whatever reason.

**Comment by seanmonstar** [comment, marked as answer]
hyper internally uses [h2](https://github.com/hyperium/h2) as a lower-level HTTP/2 implementation.

**Reply by thesw4rm** [reply]
Didn't see this reply. Any chance you can direct me to where the parsing is done for requests and responses? If not, I'll try to find it myself, but any direction you can give is helpful. Thanks!

**Reply by olix0r** [reply]
You probably want to start with the [`headers` frame type]( https://github.com/hyperium/h2/blob/3a0c622f0c9ec8f53df8f6481ba7873dc06be9cf/src/frame/headers.rs).

**Reply by thesw4rm** [reply]
This is a great starting point. Thank you!

**Reply by thesw4rm** [reply]
Follow up question. Is there a way to use `load_hpack` with pub mods to decode headers? Trying to avoid exposing the decoder itself if it isn't necessary. I'm using `unstable` feature pack where `frame` mod is public. https://github.com/hyperium/h2/blob/f6aa3be6719270cd7b4094ee1940751b5f4ec88e/src/frame/headers.rs#L208.

Also, if I can figure this out a bit more, I can try to separate the parsing logic and make it its own repo so it can be used in the future on its own., What do you think?
